### PR TITLE
nodejs-week3 warmup-calculator

### PR DESCRIPTION
**Pull request by sumanghimire79**

### CODE REVIEW

File: nodejs/week3/homework/warmup/index.js [hunk 1]
```diff
@@ -0,0 +1,183 @@
+const express = require('express');
+const app = express();
+
+// Parse URL-encoded bodies (as sent by HTML forms)
+app.use(express.urlencoded({ extended: true }));
+// Parse JSON bodies (as sent by API clients)
+app.use(express.json());
+
+// get methods:
+app.get('/calculator', (req, res) => {
```
Comment (dsamourkasidis) [inline nodejs/week3/homework/warmup/index.js:10]: While what you did works, it is too complicated to include everything in a single route. I suggest you create 4 separate routes for the addition/subtraction/multiplication/division. Eg:
```
app.get("/calculator/addition", (req, res)
```

File: nodejs/week3/homework/warmup/index.js [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+            return res.send(secondParamErrorMsg);
+        }
+        answer = Number(firstParam) / Number(secondParam);
+
+        res.json(answer);
+    }
+});
+
+//  post method
+app.post('/calculator', (req, res) => {
```
Comment (dsamourkasidis) [inline nodejs/week3/homework/warmup/index.js:97]: Nice job adding the POST request and getting the parameters from the request's body. 
However, I suggest you split them to separate routes as well:
```
app.post('/calculator/addition', (req, res)
```

**Review by dsamourkasidis — COMMENTED** [review]
Very good work 👍 
Only thing you should fix, is splitting the routes and not having all 4 features(addition/subtraction..) in the same route, have a look at my comment :)

File: nodejs/week3/homework/warmup/index.js (hunk 1 above)
Comment (sumanghimire79) [reply nodejs/week3/homework/warmup/index.js]: > Very good work 👍 Only thing you should fix, is splitting the routes and not having all 4 features(addition/subtraction..) in the same route, have a look at my comment :)

yes, it increass the same code line repeated but according to your suggestion i have splitted them into four separate routs.  thank you

Comment (sumanghimire79) [reply nodejs/week3/homework/warmup/index.js]: > While what you did works, it is too complicated to include everything in a single route. I suggest you create 4 separate routes for the addition/subtraction/multiplication/division. Eg:
> 
> ```
> app.get("/calculator/addition", (req, res)
> ```

i refactored it according to your suggestions

**Pull request merged**